corehq/apps/locations/bulk_management.py

```diff
@@ -809,12 +809,21 @@ def update_children(parent_stub):
 
         for child_stub in child_stubs:
             child = child_stub.db_object
-            child.parent = parent_location
             if child_stub.do_delete:
                 # keep track of to be deleted items to delete them in top-to-bottom order
                 to_be_deleted.append(child)
             elif child_stub.needs_save:
                 child.location_type = types_by_code.get(child_stub.location_type)
+                if parent_location:
+                    # refetch parent_location object so that mptt related fields are updated consistently,
```

Comment:
Do you mean `lineage`?  Aside from that, I wouldn't expect `rebuild` to affect any of the fields we sync between the two.